### PR TITLE
Blog post about API documentation rollout

### PR DESCRIPTION
**Pull request by saraycp**
We must inform our users right before/after we roll out the new API documentation.

Here you are the blog post I propose. Please have a look so it's ready to use when needed.

TODO:

- [x] Adapt the date in the file name and the partial.
- [ ] Publish a Status Message at the same time the blog post is merged. The text can be:

```
The [new API documentation](https://api.opensuse.org/apidocs/) is now official! Enjoy it! Read our latest [blog post](https://openbuildservice.org/2023/06/06/api-docs-rollout/) for more details.
```

### CODE REVIEW

**Review by danidoni — APPROVED** [review]

**Comment by danidoni** [comment]
LGTM

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/development/2023-06-02-api-docs-rollout.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+Our journey to remake the API documentation has come to an end 🎉
+Starting today, the documentation presented through Swagger UI using the OpenAPI definition becomes the official API documentation for Open Build Service.
+
+{% include partials/_series-of-posts-about-api-docs.md %}
+
+The URL stays the same: [https://build.opensuse.org/apidocs](https://build.opensuse.org/apidocs/index). Whereas the discontinued documentation has been moved to [https://build.opensuse.org/apidocs-old](https://build.opensuse.org/apidocs-old/index) where it will be available for a few more weeks.
+
+Our goal is to make the work easier for users and contributors. Please keep the feedback coming, it will help you and us.
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply _posts/development/2023-06-02-api-docs-rollout.md:14]: @hennevogel, we need your help here, not only to say what is left, but to write something big as you expect.

Don't hesitate to write some paragraphs about all this journey you also have experienced in this blog post. That would be extremely welcomed.

[the remark this answers is not in the document]
Comment (eduardoj) [reply _posts/development/2023-06-02-api-docs-rollout.md:14]: I want to inform our users right after we roll out the new API documentation.

We have more time: beginning next week.

[the remark this answers is not in the document]
Comment (saraycp) [reply _posts/development/2023-06-02-api-docs-rollout.md:14]: @hennevogel I could propose something. Or did you write something already??

**Comment by eduardoj** [comment]
I like it soooo much. Thanks a lot, Saray. :raised_hands: 

We could rename the file of the blog post and also the text which links to it (`_includes/partials/_series-of-posts-about-api-docs.md`) to the day of tomorrow: 2023-06-06.

File: _posts/development/2023-06-02-api-docs-rollout.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+From now on, we all can benefit from:
+
+- a more modern and maintained API documentation;
+- documentation which is easier to understand for developers who want to make use of the OBS API in their code;
+- a more standardized way to document the API, which might increase the number of contributions.
+
+To summarize, we now provide a way to create and update the OBS API, easier to read for OBS newcomers and long-time users, and easier to document for OBS developers. What a great achievement!
+
+Enjoy the new docu! 📖 The URL stays the same: [https://build.opensuse.org/apidocs](https://build.opensuse.org/apidocs/index). Whereas the discontinued documentation has been moved to [https://build.opensuse.org/apidocs-old](https://build.opensuse.org/apidocs-old/index) where it will be available for a few more weeks.
```
[the remark this answers is not in the document]
Comment (saraycp) [reply _posts/development/2023-06-02-api-docs-rollout.md:27]: So the "official" URL so far will no longer be the official one... 
It was `https://build.opensuse.org/apidocs` before but now it's `https://api.opensuse.org/apidocs/`.

[the remark this answers is not in the document]
Comment (eduardoj) [reply _posts/development/2023-06-02-api-docs-rollout.md:27]: Exactly.

[the remark this answers is not in the document]
Comment (saraycp) [reply _posts/development/2023-06-02-api-docs-rollout.md]: I rephrased the whole paragraph.

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+Starting today, the documentation presented through Swagger UI using the OpenAPI definition becomes the official API documentation for Open Build Service 🎉
+The team's journey to remake the API documentation has come to an end, but the most exciting journey is about to begin!
+One where any member of the OBS community can contribute to the OBS API documentation and use the OBS API easily and efficiently 👏
+
+{% include partials/_series-of-posts-about-api-docs.md %}
+
+With the rollout, we are moving away from:
+
+- an API documentation that wasn't very approachable for developers;
+- documentation that heavily relied on XML examples and schema files, two concepts that weren't clear enough for contributors;
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:16]: We still heavily use both no?

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+- a tool ([restility](https://github.com/openSUSE/open-build-service/tree/master/docs/api/restility)) that was no longer maintained.
+
+From now on, we all can benefit from:
+
+- a more modern and maintained API documentation;
+- documentation which is easier to understand for developers who want to make use of the OBS API in their code;
+- a more standardized way to document the API, which might increase the number of contributions.
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:23]:
```suggestion
- a more standardized way to document the API, which hopefully increase the number of contributions.
```

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: The New API Documentation Is Rolled Out!
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:3]:
```suggestion
title: OpenAPI Specification for the Open Build Service HTTP API
```

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+category: development
+---
+
+Starting today, the documentation presented through Swagger UI using the OpenAPI definition becomes the official API documentation for Open Build Service 🎉
+The team's journey to remake the API documentation has come to an end, but the most exciting journey is about to begin!
+One where any member of the OBS community can contribute to the OBS API documentation and use the OBS API easily and efficiently 👏
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:9]:
```suggestion
Your journey to get, use and contribute to the OBS API documentation! 👏
```

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: The New API Documentation Is Rolled Out!
+category: development
+---
+
+Starting today, the documentation presented through Swagger UI using the OpenAPI definition becomes the official API documentation for Open Build Service 🎉
+The team's journey to remake the API documentation has come to an end, but the most exciting journey is about to begin!
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:8]:
```suggestion
Our journey to remake the API documentation has come to an end, but the most exciting journey is about to begin.
```

File: _posts/development/2023-06-06-api-docs-rollout.md (hunk 5 above)
Comment (hennevogel) [reply _posts/development/2023-06-06-api-docs-rollout.md:3]: @eduardoj are we using a specific OpenAPI spec version?

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+One where any member of the OBS community can contribute to the OBS API documentation and use the OBS API easily and efficiently 👏
+
+{% include partials/_series-of-posts-about-api-docs.md %}
+
+With the rollout, we are moving away from:
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:13]: Rollout is a concept not many of your customers will be aware of. I wouldn't mention it.

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+- documentation that heavily relied on XML examples and schema files, two concepts that weren't clear enough for contributors;
+- a tool ([restility](https://github.com/openSUSE/open-build-service/tree/master/docs/api/restility)) that was no longer maintained.
+
+From now on, we all can benefit from:
+
+- a more modern and maintained API documentation;
+- documentation which is easier to understand for developers who want to make use of the OBS API in their code;
+- a more standardized way to document the API, which might increase the number of contributions.
+
+To summarize, we now provide a way to create and update the OBS API, easier to read for OBS newcomers and long-time users, and easier to document for OBS developers. What a great achievement!
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:25]: I find self back-patting super cheesy... Let others say what kind of achievement this is. But that might just be me... :)

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+Enjoy the new docu! 📖 The URL is [https://api.opensuse.org/apidocs/](https://api.opensuse.org/apidocs/). The discontinued documentation will be available in [https://build.opensuse.org/apidocs-old/](https://build.opensuse.org/apidocs-old/index) for a few more weeks.
```
Comment (hennevogel) [inline _posts/development/2023-06-06-api-docs-rollout.md:27]: Famous last words, I bet you this will be around for a very long time. Don't promise what you can't keep...

**Review by hennevogel — CHANGES_REQUESTED** [review]
I think you are missing a bit more explanation of how to use this whole thing from the customer perspective. [swagger-ui](https://swagger.io/tools/swagger-ui/) is not super self-explanatory and neither is contributing to [OBS-v2.10.50.yaml](https://api.opensuse.org/apidocs-new/OBS-v2.10.50.yaml)

**Comment by saraycp** [comment]
@hennevogel I address all your feedback. Only waiting for the answer to

> @eduardoj are we using a specific OpenAPI spec version?

to specify the version number if needed.

**Comment by eduardoj** [comment]
> @hennevogel I address all your feedback. Only waiting for the answer to
> 
> > @eduardoj are we using a specific OpenAPI spec version?
> 
> to specify the version number if needed.

Yes, we are. It's defined here:

https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs-new/OBS-v2.10.50.yaml#L1

**Comment by saraycp** [comment]
> I think you are missing a bit more explanation of how to use this whole thing from the customer perspective. [swagger-ui](https://swagger.io/tools/swagger-ui/) is not super self-explanatory and neither is contributing to [OBS-v2.10.50.yaml](https://api.opensuse.org/apidocs-new/OBS-v2.10.50.yaml)
> 
> Yet those are the two things you want people to do after reading this blog post right?

I'm with this now..

**Comment by saraycp** [comment]
@eduardoj @hennevogel please have another look.

I added two sections talking about how to use the docu and how to contribute.

I finally mentioned the OpenAPI version in the content, not in the title:

> First of all, you need to understand what is Swagger UI and learn the OpenAPI 3.0.0 Specification.

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+### Want To Contribute?
+
+First of all, you need to understand what is Swagger UI and learn the OpenAPI 3.0.0 Specification. [This OpenAPI Guide](https://swagger.io/docs/specification/about/) is a good starting point.
+
+Once you are familiar with the OpenAPI specification structure, you can start contributing to OBS documentation, which you can find under [src/api/public/apidocs](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs) in our GitHub repository.
+
+Have a look to [closed pull requests](https://github.com/openSUSE/open-build-service/pulls?q=is%3Apr+label%3A%22Documentation+%3Abook%3A%22+is%3Aclosed) to take what others did before as an example. The main steps are:
+
+- Declare the new endpoint in [OBS-v2.10.50.yaml](https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs/OBS-v2.10.50.yaml). Please keep the alphabetical order.
```
Comment (eduardoj) [inline _posts/development/2023-06-06-api-docs-rollout.md:50]: A contribution to the documentation will not necessarily mean adding an endpoint. I would rephrase this as:
```suggestion
- If you are documenting a new endpoint, add it in [OBS-v2.10.50.yaml](https://github.com/openSUSE/open-build-service/blob/master/src/api/public/apidocs/OBS-v2.10.50.yaml). Please keep the alphabetical order.
```

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+- Define the endpoint in YAML format under the [paths/](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs/paths) directory.
```
Comment (eduardoj) [inline _posts/development/2023-06-06-api-docs-rollout.md:51]: Also here, contributions will not always mean to define new endpoints. I also find `in YAML format` kind of redundant. I would use something like:
```suggestion
- Define or modify how endpoints behave under the [paths/](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs/paths) directory.
```

File: _posts/development/2023-06-06-api-docs-rollout.md (hunk 11 above)
Comment (danidoni) [reply _posts/development/2023-06-06-api-docs-rollout.md:50]: I think the correct way is `add it to ...`

Comment (hennevogel) [reply _posts/development/2023-06-06-api-docs-rollout.md:50]: > Please keep the alphabetical order.

I think this is too much information.

File: _posts/development/2023-06-06-api-docs-rollout.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+- Make use of the already-defined [components](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs/components) (parameters, responses, schemas) when possible.
+
+You can also have a look to [closed pull requests](https://github.com/openSUSE/open-build-service/pulls?q=is%3Apr+label%3A%22Documentation+%3Abook%3A%22+is%3Aclosed) to take what others did before as an example.
+
+Enjoy the new docu 📖 and keep the feedback and contributions coming, please. It will help the whole OBS community.
+
+{% include partials/_how-to-give-us-feedback.md %}
+
+And a wholeheartedly thank you to everyone that has contributed and gave us feedback during the development of this.
```
Comment (eduardoj) [inline _posts/development/2023-06-06-api-docs-rollout.md:49]:
```suggestion
And a wholeheartedly thank you to everyone that has created the initial API documentation, and later on, contributed and gave us feedback during the development of this new documentation.
```

I would also mention the people who created the initial documentation, which served us as base to create the new one.

Also, finishing the sentence with "... of this." sounded weird.

**Review by eduardoj — CHANGES_REQUESTED** [review]
Thanks for the suggestions! Just a couple of minor changes I found we should need, in the "thank you" section.

File: _posts/development/2023-06-06-api-docs-rollout.md (hunk 13 above)
Comment (eduardoj) [reply _posts/development/2023-06-06-api-docs-rollout.md]: By "initial API documentation" I mean "now discontinued API documentation". I will modify this suggestion in another comment.

File: _posts/development/2023-06-06-openapi-documentation.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+- Define the endpoint in YAML format under the [paths/](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs/paths) directory.
+- Make use of the already-defined [components](https://github.com/openSUSE/open-build-service/tree/master/src/api/public/apidocs/components) (parameters, responses, schemas) when possible.
+
+You can also have a look to [closed pull requests](https://github.com/openSUSE/open-build-service/pulls?q=is%3Apr+label%3A%22Documentation+%3Abook%3A%22+is%3Aclosed) to take what others did before as an example.
+
+Enjoy the new docu 📖 and keep the feedback and contributions coming, please. It will help the whole OBS community.
+
+{% include partials/_how-to-give-us-feedback.md %}
+
+And a wholeheartedly thank you to everyone that has contributed and gave us feedback during the development of this new documentation.
```
Comment (eduardoj) [inline _posts/development/2023-06-06-openapi-documentation.md:48]:
```suggestion
A wholehearted thank you to everyone that created the now discontinued API documentation, and also to those who have contributed and given us feedback during the development of this new documentation.
```

Comment (saraycp) [reply _posts/development/2023-06-06-openapi-documentation.md:48]: It's clearer now. You meant "the former API" and not our first contributions to the new one. I misunderstood before, sorry. Adding the change...
